### DISCUSSION TITLE
Getting parser confidence

### DISCUSSION THREAD

**Discussion by BramVanroy** [explosion/spaCy#13340] [Help: Coding & Implementations] [answered]
Hello, been a while!

I would like to get the parser's confidence after parsing (or calculate its perplexity for a given dependency tree), to use it for reranking some generated sentences based on their syntactic structure. 

[A long time ago](https://github.com/explosion/spaCy/issues/881), @honnibal worked on this and `beam_parse` was added in v2. This enabled a way to retrieve NER confidence scores. AFAICT this has not persisted in v3 but, important for me, dependency parsing has not received this API.

So, although I fear I already know the answer, I'm asking whether there is any way to retrieve a parser's confidence scores after parsing (or its PPL for a given dependency tree). Thanks!

**Comment by svlandeg** [comment, marked as answer]
Hey Bram, I think your question got answered on [Twitter](https://twitter.com/BramVanroy/status/1759895160165487017), or did you have any other doubts?

We've actually been working on the beam parsing, but still haven't been able to pull it out of experimental status yet...

**Reply by BramVanroy** [reply]
Indeed, thanks for the replies there, and posting the link here so others can find it!
